### PR TITLE
feat: add itinerary checkboxes

### PR DESCRIPTION
**Pull request by Jeffrey-Zutt**
Adds checkboxes to itineray items

### CODE REVIEW

File: src/components/itineraries/DraggableItineraryItem.tsx [hunk 1]
```diff
@@ -62,6 +70,8 @@ const DraggableItineraryItem: FC<Props> = ({ itineraryItem, index }) => {
     }
   )
 
+  const onCheck= () => setChecked(id, !checked)
```
Comment (gilleswittenberg) [inline src/components/itineraries/DraggableItineraryItem.tsx:73]: space after `onCheck`. Let's add this to ESLint.

File: src/components/itineraries/SpinnerCheckbox.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+const SIZE = WIDTH - PADDING*2 - MARGIN*2 - BORDER_WIDTH*2
+
+const StyledSpinner = styled(Spinner)`  
+  padding: ${PADDING}px;
+  border: ${BORDER_WIDTH}px solid black;  
+  margin: ${MARGIN}px;
+`
+
+const SpinnerCheckbox:React.FC<Props> = ({ onChange, checked }) => {
+  const [ isLoading, setIsLoading ] = useState<boolean>(false)
```
Comment (gilleswittenberg) [inline src/components/itineraries/SpinnerCheckbox.tsx:27]: Type `boolean` will be inferred

File: src/components/itineraries/DraggableItineraryItem.tsx (hunk 1 above)
Comment (Jeffrey-Zutt) [reply src/components/itineraries/DraggableItineraryItem.tsx]: Linter now catches these things

File: src/components/itineraries/SpinnerCheckbox.tsx (hunk 2 above)
Comment (Jeffrey-Zutt) [reply src/components/itineraries/SpinnerCheckbox.tsx]: Ah good one.

**Pull request merged**